### PR TITLE
Adding-check-domain-is-not-available-for-producer-consumer-in-create

### PR DESCRIPTION
**Pull request by sunlnu**
# stream-registry PR

_&lt;Producer consumer create fails when domain is not exist&gt;_

### Added
* _&lt;Producer consumer create fails when domain is not exist&gt;_
* _&lt;Producer consumer create fails when domain is not exist&gt;_

### Changed
* _&lt;Producer consumer validation implementation&gt;_

### Deleted
* _&lt;only added&gt;_

# PR Checklist Forms

- [ ] CHANGELOG.md updated
- [ ] Reviewer assigned
- [ ] PR assigned (presumably to submitter)
- [ ] Labels added (enhancement, bug, documentation)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [unreleased]
+### Added
+- producer & consumer creation not allowed when domain is not available
+
```
Comment (techabstraction) [inline CHANGELOG.md:10]: While you are touching this code perhaps you could fix this also? 

Renaming `stream` field in the classes `ProcessInputStream` and `ProcessOutputStream`:
`StreamKey stream;` -> `StreamKey streamKey;`

Renaming `stream` field in the classes `ProcessInputStreamInput` and `ProcessOutputStreamInput`:
`StreamKeyInput stream;` -> `StreamKeyInput streamKey;`

```suggestion
### Changed
- Fixed `streamKey` field name in `ProcessInputStream` and `ProcessOutputStream`
- Fixed `streamKey` field name in `ProcessInputStreamInput` and `ProcessOutputStreamInput`
```

**Review by techabstraction — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (sunlnu) [reply CHANGELOG.md:12]: done .

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [unreleased]
+### changed
```
Comment (techabstraction) [inline CHANGELOG.md:8]: Minor:
```suggestion
### Changed
```

Comment (sunlnu) [reply CHANGELOG.md]: done

File: graphql/api/src/main/java/com/expediagroup/streamplatform/streamregistry/graphql/model/inputs/ProcessInputStreamInput.java [hunk 3]
```diff
@@ -25,10 +25,10 @@
 @Value
 @Builder
 public class ProcessInputStreamInput {
-  StreamKeyInput stream;
+  StreamKeyInput streamKeyInput;
```
Comment (techabstraction) [inline graphql/api/src/main/java/com/expediagroup/streamplatform/streamregistry/graphql/model/inputs/ProcessInputStreamInput.java:28]: This seems to be the convention we are following in the other input classes?
```suggestion
  StreamKeyInput streamKey;
```

File: graphql/api/src/main/java/com/expediagroup/streamplatform/streamregistry/graphql/model/inputs/ProcessOutputStreamInput.java [hunk 4]
```diff
@@ -25,10 +25,10 @@
 @Value
 @Builder
 public class ProcessOutputStreamInput {
-  StreamKeyInput stream;
+  StreamKeyInput streamKeyInput;
```
Comment (techabstraction) [inline graphql/api/src/main/java/com/expediagroup/streamplatform/streamregistry/graphql/model/inputs/ProcessOutputStreamInput.java:28]:
```suggestion
  StreamKeyInput streamKey;
```

File: graphql/api/src/main/resources/stream-registry.graphql [hunk 5]
```diff
@@ -300,12 +300,12 @@ input ProcessInputStreamBindingInput {
...
 
 type ProcessOutputStream {
-    stream: StreamKey!
+    streamKey: StreamKey!
     configuration: ObjectNode!
 }
 
 input ProcessOutputStreamInput {
-    stream: StreamKeyInput!
+    streamKeyInput: StreamKeyInput!
```
Comment (techabstraction) [inline graphql/api/src/main/resources/stream-registry.graphql:308]:
```suggestion
    streamKey: StreamKeyInput!
```

File: graphql/api/src/main/resources/stream-registry.graphql [hunk 6]
```diff
@@ -280,12 +280,12 @@ type Consumer {
...
 
 type ProcessInputStream {
-    stream: StreamKey!
+    streamKey: StreamKey!
     configuration: ObjectNode!
 }
 
 input ProcessInputStreamInput {
-    stream: StreamKeyInput!
+    streamKeyInput: StreamKeyInput!
```
Comment (techabstraction) [inline graphql/api/src/main/resources/stream-registry.graphql:288]: This seems to be the convention elsewhere in this file?
```suggestion
    streamKey: StreamKeyInput!
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [unreleased]
+### Changed
+- producer & consumer creation not allowed when domain is not available
+- Fixed `streamKey` field name in `ProcessInputStream` and `ProcessOutputStream`
+- Fixed `streamKey` field name in `ProcessInputStreamInput` and `ProcessOutputStreamInput`
+
```
Comment (techabstraction) [inline CHANGELOG.md:12]: Just spotted a couple more in the process model:
```suggestion
- Fixed `zoneKeys` field in `Process`
- Fixed `zoneKey` field in `ProcessBinding`

```

**Pull request merged**